numpy/f2py/src/fortranobject.c

```diff
@@ -691,7 +696,7 @@ PyArrayObject* array_from_pyobj(const int type_num,
         }
         arr = (PyArrayObject *)
             PyArray_New(&PyArray_Type, rank, dims, type_num,
-                        NULL,NULL,0,
+                        NULL,NULL,itemsize,
```

Comment:
I think you can just set this to `1`. It is ignored for types of fixed size, which should be the case for NPY_CHAR and all the numeric types,  so only used for string types